pkg/common/utils.go

```diff
@@ -148,3 +148,36 @@ func CreateNodeID(project, zone, name string) string {
 func CreateZonalVolumeID(project, zone, name string) string {
 	return fmt.Sprintf(volIDZonalFmt, project, zone, name)
 }
+
+// ConvertLabelsStringToMap converts the labels from string to map
+// example: "key1=value1,key2=value2" gets converted into {"key1": "value1", "key2": "value2"}
+func ConvertLabelsStringToMap(labels string) (map[string]string, error) {
+	const labelsDelimiter = ","
+	const labelsKeyValueDelimiter = "="
+
+	m := make(map[string]string)
+	if labels == "" {
+		return m, nil
+	}
+
+	errIncorrectFormat := fmt.Errorf("labels %q are invalid, correct format: 'key1=value1,key2=value2'", labels)
+
+	keyValueStrings := strings.Split(labels, labelsDelimiter)
+	for _, keyValue := range keyValueStrings {
+		keyValue := strings.Split(keyValue, labelsKeyValueDelimiter)
+
+		if len(keyValue) != 2 {
+			return nil, errIncorrectFormat
+		}
+
+		key := strings.TrimSpace(keyValue[0])
+		if key == "" {
+			return nil, errIncorrectFormat
+		}
+
```

Comment:
Are empty values ok?

Reply:
Yes, according to the link you shared above:

_Keys have a minimum length of 1 character and a maximum length of 63 characters, and cannot be empty. Values can be empty, and have a maximum length of 63 characters._

Reply:
Well at least I found the link :-P

Thanks